content/docs/user-guide/assets/pipeline/asset-builders.md

```diff
@@ -1,71 +1,190 @@
+* The referenced asset may not be on the same drive as the asset with the reference. This includes scan directories, which can be on different drives. This means relative paths may not be resolvable.
+* Some operating systems have case sensitive paths, some do not. When resolving paths in an asset builder, it's important to do so in a way that will be stable across operating systems.
+* File locations relative to scan directories, and in the asset cache will be stable for all members of a team, but scan directories themselves, and files outside scan directories may not have the same paths for all team members.
+* Path resolving logic should be consistent, and predictable. Content creators will want to understand how paths are resolved when processing assets, so they can better manage their content.
+
+### References from product assets to other files
+
+![Product asset references](/images/user-guide/assets/pipeline/asset_builders/product_asset_references.png)
+
+Product assets should only have these references:
+* Product assets by asset ID
+  * This is the prefered way to reference other product assets, references by path should be avoided when possible
+* Product assets by a relative path from the asset with the reference
+* Product assets by a relative path, to a specific root in the asset cache, usually the cache root for the platform
+* Non-product file in the project directory, expected at a specific location
+  * This kind of reference is not common, but there are cases where it may be necessary to reference a file outside the asset folder that will be in the shipped final product.
+
+While it's technically possible to reference files in other ways in product assets, it may end up causing problems later. The development version of a project, running the Editor or game launcher connected to the Asset Processor is what a team will be most used to interacting with. However, when preparing to deliver a project to end customers, with packaged and bundled content, the layout of the project and placement of assets will be different. References from product assets to other files that worked during development may not work in release builds of projects. For example, if a product asset references a source asset, that source asset likely won't be on the end user's machine in the packaged release build.
+
+Product assets should never reference other product assets with absolute path because this will result in the hash of the contents of product assets being unique per machine that generates product assets. When authoring a builder, the complete lifecycle of asset management for a project should be kept in mind. Keeping product assets stable across machines will ensure that the gathering modified assets step of generating a patch for a live game is accurate. You can read more about [asset bundling here.](docs/user-guide/packaging/asset-bundler/overview/)

```

Comment:
Broken link
```suggestion
Product assets should never reference other product assets with absolute path because this will result in the hash of the contents of product assets being unique per machine that generates product assets. When authoring a builder, the complete lifecycle of asset management for a project should be kept in mind. Keeping product assets stable across machines will ensure that the gathering modified assets step of generating a patch for a live game is accurate. You can read more about [asset bundling here.](/docs/user-guide/packaging/asset-bundler/overview/)

```

Reply:
Fixed in next commit.